pokemon.go

```diff
@@ -51,3 +86,46 @@ func (p *Pokemon) String() string {
 	return fmt.Sprintf("%v%v\tLv%d\nHP: %d/%d\n", p.GetName(),
 		p.Gender, p.Level, p.CurrentHP, p.Stats[0])
 }
+
+// Restore HP to a Pokemon. Can also be used to revive a fainted Pokemon.
+func (p *Pokemon) RestoreHP(amount uint) {
+	if p.Stats[STAT_HP]-p.CurrentHP <= amount {
+		p.CurrentHP = p.Stats[STAT_HP]
+	} else {
+		p.CurrentHP += amount
+	}
+}
+
+// Cures a status ailment from a Pokemon.
+func (p *Pokemon) CureStatusEffect(status uint) {
+	p.StatusEffects &= ^status
+}
+
+// Modifies one of the six base stats for a Pokemon
+func (p *Pokemon) ModifyStat(stat, stages int) {
```

Comment:
(nit)
```suggestion
func (p *Pokemon) BoostStat(stat, stages int) {
```

Reply:
Is boost ambigious because this can take negative stages for a debuff?

Reply:
Oh, you're right, maybe it is. I think it should be something other than modify, because it makes me think it modifies the underlying stat, not adding buffs/debuffs

Reply:
https://bulbapedia.bulbagarden.net/wiki/Stat#Stat_modifiers Seems like modify is the word.